src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -2293,6 +2295,49 @@ internal void OnBeforeDeleteInvoke(BeforeDeleteEventArgs beforeDeleteEventArgs)
         {
             OnBeforeDelete?.Invoke(this, beforeDeleteEventArgs);
         }
+
+        internal StreamResult<T> CreateStreamResult<T>(
+            BlittableJsonReaderObject json,
+            FieldsToFetchToken fieldsToFetch, 
+            bool isProjectInto)
+        {
+            var metadata = json.GetMetadata();
+            var changeVector = metadata.GetChangeVector();
+            //MapReduce indexes return reduce results that don't have @id property
+            metadata.TryGetId(out string id);
+
+            return new StreamResult<T>
+            {
+                ChangeVector = changeVector,
+                Id = id,
+                Metadata = new MetadataAsDictionary(metadata),
+                Document = QueryOperation.Deserialize<T>(id, json, metadata, fieldsToFetch, true, this, isProjectInto)
+            };
+        }
+
+        internal TimeSeriesStreamResult<T> CreateTimeSeriesStreamResult<T>(StreamOperation.YieldStreamResults enumerator) where T : new()
+        {
+            var json = enumerator.Current;
+            var metadata = json.GetMetadata();
+            var changeVector = metadata.GetChangeVector();
+            //MapReduce indexes return reduce results that don't have @id property
+            metadata.TryGetId(out string id);
+
+            var result = new TimeSeriesStreamResult<T>
+            {
+                ChangeVector = changeVector, 
+                Id = id, 
+                Metadata = new MetadataAsDictionary(metadata), 
+                
+                //TODO: replace this?
+                Result = new T()
+            };
+            
+            var o = result.Result as ITimeSeriesQueryStreamResult;
```

Comment:
what if this is null? Shouldn't we change 'where' to ` where T : ITimeSeriesQueryStreamResult, new()` ?